akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala

```diff
@@ -29,25 +31,24 @@ object PersistentActorCompileOnlyTest {
     //#state
 
     //#command-handler
-    val commandHandler: CommandHandler[SimpleCommand, SimpleEvent, ExampleState] =
-      CommandHandler.command {
+    def commandHandler(state: ExampleState): CommandHandler[SimpleCommand, SimpleEvent, ExampleState] =
+      command {
         case Cmd(data) ⇒ Effect.persist(Evt(data))
       }
     //#command-handler
 
     //#event-handler
-    val eventHandler: (ExampleState, SimpleEvent) ⇒ (ExampleState) = {
-      case (state, Evt(data)) ⇒ state.copy(data :: state.events)
+    def eventHandler(state: ExampleState): EventHandler[SimpleEvent, ExampleState] = {
+      case Evt(data) ⇒ state.copy(data :: state.events)
     }
     //#event-handler
 
     //#behavior
     val simpleBehavior: PersistentBehavior[SimpleCommand, SimpleEvent, ExampleState] =
-      PersistentBehaviors.receive[SimpleCommand, SimpleEvent, ExampleState](
-        persistenceId = "sample-id-1",
-        initialState = ExampleState(Nil),
-        commandHandler = commandHandler,
-        eventHandler = eventHandler)
+      PersistentBehaviors[SimpleCommand, SimpleEvent, ExampleState]
+        .identifiedBy("sample-id-1")
```

Comment:
I didn't think about this option. The previous API had a `receive` method so I was trying to replace it by some more obvious.

The apply method is a good option if the typed are already defined on the `val` declaration.  Otherwise, we will need to right:

```scala
PersistentBehavior[SimpleCommand, SimpleEvent, ExampleState](persistenceId = "sample-id-1")
```
That doesn't read nice, but I guess it's preferable to declare the types on the `val`.